lib/ansible/plugins/connection/winrm.py

```diff
@@ -86,6 +87,15 @@ def __init__(self,  *args, **kwargs):
 
         super(Connection, self).__init__(*args, **kwargs)
 
+        self.host = self._play_context.remote_addr
```

Comment:
Do these values even need to exist in most of the plugins? Only reason I can see that they'd need to is if `transport_test` were defined on the base connection (and override-able by plugins that don't use a TCP port). Otherwise, `transport_test` can just use the existing `self._winrm_(host/port)`. If they do stay, I'd prefer that these values be init'd in `set_host_overrides` off the existing values instead of duplicating the logic here (at least for this plugin- IIRC nobody else uses `set_host_overrides` ATM). I've verified that it works with that change (since you're using the play's existing connection, it's already been run by the time it gets there).